### ISSUE TITLE
Docker Container Creation Error - Access to the path 'C:\Run\Collation.txt' is denied.

### ISSUE THREAD

**Issue by jddevBC** [microsoft/navcontainerhelper#2414]
I am receiving this error while attempting to run a basic  container creation script.

I have tried:
     1.  adding -isolation hyperv
     2.  removing -updatehosts
Neither worked.  Any ideas on how to fix this issue?  I did create a new container 2 days ago that I connected to an external DB without issue.  But, this attempt to create a basic container connected to the express Cronus database is not working.

Docker version: 
PS C:\WINDOWS\system32> docker version
Client:
 Cloud integration: v1.0.23
 Version:           20.10.14
 API version:       1.41
 Go version:        go1.16.15
 Git commit:        a224086
 Built:             Thu Mar 24 01:53:11 2022
 OS/Arch:           windows/amd64
 Context:           default
 Experimental:      true

Server: Docker Desktop 4.7.1 (77466)
 Engine:
  Version:          20.10.14
  API version:      1.41 (minimum version 1.24)
  Go version:       go1.16.15
  Git commit:       87a90dc
  Built:            Thu Mar 24 01:48:25 2022
  OS/Arch:          windows/amd64
  Experimental:     false

BCContainerHelper Version: 3.0.5

![image](https://user-images.githubusercontent.com/26700420/163593904-bd4c0521-caf5-4cf7-b200-b4c0319ac815.png)

BcContainerHelper is version 3.0.5
BcContainerHelper is running as administrator
Host is Microsoft Windows 10 Pro - 21H1
Docker Client Version is 20.10.14
Docker Server Version is 20.10.14
Removing container containername
Removing containername from container hosts file
Removing containername-* from container hosts file
Removing C:\ProgramData\BcContainerHelper\Extensions\containername
Fetching all docker images
Fetching all docker volumes
INFO: Windows 10 21H1/21H2 images are not yet available, using 2004 as these are found to work better than 20H2 on 21H1/21H2
Using image mcr.microsoft.com/businesscentral:10.0.19041.1415
Creating Container containername
Style: onprem
Multitenant: No
Version: 18.12.38101.0
Platform: 18.0.38042.0
Generic Tag: 1.0.2.1
Container OS Version: 10.0.19041.1415 (2004)
Host OS Version: 10.0.19043.1526 (21H1)
WARNING: Host OS is 21H1 and Container OS is 2004, defaulting to process isolation. If you experience problems, add -isolation hyperv.
Using process isolation
Using locale en-US
Disabling the standard eventlog dump to container log every 2 seconds (use -dumpEventLog to enable)
Using license file C:\name\License\D365BC19.flf
Files in C:\ProgramData\BcContainerHelper\Extensions\containername\my:
- AdditionalOutput.ps1
- license.flf
- MainLoop.ps1
- SetupNavUsers.ps1
- SetupVariables.ps1
- updatecontainerhosts.ps1
Creating container containername from image mcr.microsoft.com/businesscentral:10.0.19041.1415
dc65c1f5e17ecfe419a0e798ba3c9b4ded9354107935f5e885e4db2c7734204e
Waiting for container containername to be ready
Using artifactUrl https://bcartifacts.azureedge.net/onprem/18.12.38101.0/us
Using installer from C:\Run\150-new
Installing Business Central
Installing from artifacts
Starting Local SQL Server
Starting Internet Information Server
Copying Service Tier Files
c:\dl\onprem\18.12.38101.0\platform\ServiceTier\Program Files
c:\dl\onprem\18.12.38101.0\platform\ServiceTier\System64Folder
Copying PowerShell Scripts
c:\dl\onprem\18.12.38101.0\platform\WindowsPowerShellScripts\Cloud\NAVAdministration
c:\dl\onprem\18.12.38101.0\platform\WindowsPowerShellScripts\WebSearch
Copying dependencies
Copying ReportBuilder
Importing PowerShell Modules
Determining Database Collation from c:\dl\onprem\18.12.38101.0\us\database\Demo Database NAV (18-0).bak
Changing Database Server Collation to Latin1_General_100_CI_AS
Access to the path 'C:\Run\Collation.txt' is denied.
at SetDatabaseServerCollation, C:\Run\HelperFunctions.ps1: line 629
at <ScriptBlock>, C:\Run\navinstall.ps1: line 222
at <ScriptBlock>, C:\Run\start.ps1: line 225
at <ScriptBlock>, <No file>: line 1Error
Using artifactUrl https://bcartifacts.azureedge.net/onprem/18.12.38101.0/us
Using installer from C:\Run\150-new
Installing Business Central
Installing from artifacts
Starting Local SQL Server
Starting Internet Information Server
Copying Service Tier Files
c:\dl\onprem\18.12.38101.0\platform\ServiceTier\Program Files
c:\dl\onprem\18.12.38101.0\platform\ServiceTier\System64Folder
Copying PowerShell Scripts
c:\dl\onprem\18.12.38101.0\platform\WindowsPowerShellScripts\Cloud\NAVAdministration
c:\dl\onprem\18.12.38101.0\platform\WindowsPowerShellScripts\WebSearch
Copying dependencies
Copying ReportBuilder
Importing PowerShell Modules
Determining Database Collation from c:\dl\onprem\18.12.38101.0\us\database\Demo Database NAV (18-0).bak
Changing Database Server Collation to Latin1_General_100_CI_AS
Access to the path 'C:\Run\Collation.txt' is denied.
at SetDatabaseServerCollation, C:\Run\HelperFunctions.ps1: line 629
at <ScriptBlock>, C:\Run\navinstall.ps1: line 222
at <ScriptBlock>, C:\Run\start.ps1: line 225
at <ScriptBlock>, <No file>: line 1
New-BcContainer Telemetry Correlation Id: a1ec67f2-7027-4efb-a605-87c5ff81efa2
Initialization of container containername failed
At C:\Program Files\WindowsPowerShell\Modules\BcContainerHelper\3.0.5\ContainerHandling\Wait-NavContainerReady.ps1:42 char:17
+ ...             throw "Initialization of container $containerName failed" ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : OperationStopped: (Initialization ...oyotaaut failed:String) [], RuntimeException
    + FullyQualifiedErrorId : Initialization of container containername failed

Event Viewer is displaying the following error:

Error running exec a5feb1c6cbeaa381d7eaf85e12c89312f77454622612623fd48bf1f7a2a7cb3a in container: container 0417602b50c444413b059abc6bb7e2163edff9f9963334c20896e8555944a1b8 encountered an error during hcsshim::System::CreateProcess: failure in a Windows system call: The user name or password is incorrect. (0x52e)

**Comment by freddydk** [comment]
Could you share the output with -isolation hyperv?

**Comment by kannanmdu** [comment]
<img width="818" alt="image" src="https://user-images.githubusercontent.com/45143585/163653720-812b7851-9509-4d98-bedc-242626632dad.png">

**Comment by freddydk** [comment]
As you can see, the color of the -isolation hyperv line is different from the other parameters.
This is because the preceding line wasn't terminated by a ` character, meaning that you are not using hyperv - the command is terminated earlier.
This blog post talks about some of the common mistakes in PowerShell: https://freddysblog.com/2019/08/04/powershell-for-non-experts/

**Comment by jddevBC** [comment]
Hi Freddy.  Thank you for the response.  The screenshot above was not mine.  I attempted to use the -isolation hyperv in my script more than once when.  I do understand the off-colored syntax.  I do not have a screenshot from my attempts to run the script.

However, I was able to get past this issue.  I uninstalled docker.  After re-installing, I did not receive the errors.  

Thank you for looking into this.
